src/configSchema.ts

```diff
@@ -0,0 +1,207 @@
+import { sep } from "path";
+
+/**
+ * error messages and map returned upon Zod validation failure
+ */
+const ZodErrorMessages = {
+	vercel: "format does not follow vercel's `ms` style ( https://github.com/vercel/ms#examples )",
+	emptyString:
+		"cannot have an empty string. If you want to unset it, use null or undefined.",
+	delay: "delay is in seconds, you can't travel back in time.",
+	fuzzySizeThreshold: "fuzzySizeThreshold must be between 0 and 1.",
+	injectUrl:
+		"You need to specify rtorrentRpcUrl, transmissionRpcUrl, qbittorrentUrl, or delugeRpcUrl when using 'inject'",
+	dataBased:
+		"Data-Based Matching requires linkType, dataDirs, and linkDir to be defined.",
+	riskyRecheckWarn:
+		"It is strongly recommended to not skip rechecking for risky matching mode.",
+	windowsPath: `Your path is not formatted properly for Windows. Please use "\\\\" or "/" for directory separators.`,
+};
+
+/**
+ * custom zod error map for logging
+ * @param error ZodIssue messaging object
+ * @param ctx ZodError map
+ * @returns (the custom error for config display)
+ */
+export const zodErrorMap: z.ZodErrorMap = (error, ctx) => {
+	switch (error.code) {
+		case z.ZodIssueCode.invalid_union:
+			return {
+				message: error.unionErrors
+					.reduce((acc, error) => {
+						// @ts-expect-error avoids type conflict string to never
+						error.errors.forEach((x) => acc.push(x.message));
+						return acc;
+					}, [])
+					.join("; "),
+			};
+	}
+
+	return { message: ctx.defaultError };
+};
+
+/**
+ * adds an issue in Zod's error mapped formatting
+ * @param setting the value of the setting
+ * @param errorMessage the error message to append on a newline
+ * @param ctx ZodError map
+ */
+function zodAddIssue(setting: string, errorMessage: string, ctx): void {
+	ctx.addIssue({
+		code: "custom",
+		message: `Setting: "${setting}"\n\t${errorMessage}`,
+	});
+}
+
+/**
+ * helper function for ms time validation
+ * @return transformed duration (string -> milliseconds)
+ */
+
+function transformDurationString(durationStr: string, ctx) {
+	const duration = ms(durationStr);
+	if (isNaN(duration)) {
+		// adds the error to the Zod Issues
+		zodAddIssue(durationStr, ZodErrorMessages.vercel, ctx);
+	}
+	return duration;
+}
+
+/**
+ * helper function for directory validation
+ * @return true if valid formatting
+ */
+function checkValidPathFormat(path: string, ctx) {
+	if (sep === "\\" && !path.includes("\\") && !path.includes("/")) {
+		zodAddIssue(path, ZodErrorMessages.windowsPath, ctx);
+	}
+	return path;
+}
+
+/**
+ * an object of the zod schema
+ * each are named after what they are intended to validate
+ */
+
+export const VALIDATION_SCHEMA = z
+	.object({
+		delay: z.number().positive({
+			message: ZodErrorMessages.delay,
+		}),
+		torznab: z.array(z.string().url()),
+		dataDirs: z
+			.array(
+				z
+					.string()
+					.transform((value, ctx) =>
+						value && value.length > 0
+							? checkValidPathFormat(value, ctx)
+							: null
```

Comment:
I'm not sure I understand what this ternary does now

Reply:
Exactly